### PR TITLE
feature: support X-RateLimit-Reset-After

### PR DESCRIPTION
**Pull request by foxbot**
Users may now optionally disable using the system clock to calculate
the ratelimit duration. This may be overrided globally, via
DiscordConfig, or per RequestOptions.

This change has been built and tested via the integrated test suite,
but has not been tested in the real world. Please verify this does not
break any of the edge-case ratelimits.

### CODE REVIEW

File: src/Discord.Net.Rest/DiscordRestApiClient.cs [hunk 1]
```diff
@@ -46,12 +46,13 @@ internal class DiscordRestApiClient : IDisposable
...
         internal ulong? CurrentUserId { get; set; }
         public RateLimitPrecision RateLimitPrecision { get; private set; }
+		internal bool UseSystemClock { get; set; }
         
         internal JsonSerializer Serializer => _serializer;
 
         /// <exception cref="ArgumentException">Unknown OAuth token type.</exception>
         public DiscordRestApiClient(RestClientProvider restClientProvider, string userAgent, RetryMode defaultRetryMode = RetryMode.AlwaysRetry,
-            JsonSerializer serializer = null, RateLimitPrecision rateLimitPrecision = RateLimitPrecision.Second)
+            JsonSerializer serializer = null, RateLimitPrecision rateLimitPrecision = RateLimitPrecision.Second, bool useSystemClock = false)
```
Comment (Chris-Johnston) [inline src/Discord.Net.Rest/DiscordRestApiClient.cs]: Unused parameter, looks like it was meant to set the `UseSystemClock` property?

**Review by Chris-Johnston — CHANGES_REQUESTED** [review]

**Review by Chris-Johnston — APPROVED** [review]
